src/net_processing.cpp

```diff
@@ -1889,7 +1890,13 @@ void PeerManagerImpl::RelayTransaction(const uint256& txid)
     CInv inv(::dstxManager->GetDSTX(txid) ? MSG_DSTX : MSG_TX, txid);
     m_connman.ForEachNode([&inv](CNode* pnode)
     {
-        pnode->PushInventory(inv);
+        LOCK(pnode->m_tx_relay->cs_tx_inventory);
+        if (pnode->m_tx_relay->filterInventoryKnown.contains(inv.hash)) {
+            LogPrint(BCLog::NET, "%s -- skipping known inv: %s peer=%d\n", __func__, inv.ToString(), pnode->GetId());
+            return;
+        }
+        LogPrint(BCLog::NET, "%s -- adding new inv: %s peer=%d\n", __func__, inv.ToString(), pnode->GetId());
+        pnode->m_tx_relay->setInventoryTxToSend.insert(inv.hash);
```

Comment:
19347: I don't like this code duplication tbh... let's maybe do 43c06d73 instead?